test/package_analysis.dart

```diff
@@ -0,0 +1,81 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:analyzer/dart/analysis/analysis_context_collection.dart';
+import 'package:analyzer/file_system/overlay_file_system.dart';
+import 'package:analyzer/file_system/physical_file_system.dart';
+import 'package:pana/src/package_analysis/common.dart';
+import 'package:pana/src/package_analysis/summary.dart';
+import 'package:path/path.dart' as path;
+import 'package:test/test.dart';
+import 'package:yaml/yaml.dart';
+
+Future<void> main() async {
+  final yamlDir =
+      Directory(path.join(path.current, 'test', 'package_analysis'));
+  final configTemplate = '''{
+  "configVersion": 2,
+  "packages": [
+    {
+      "name": "test_package",
+      "rootUri": "../",
+      "packageUri": "lib/",
+      "languageVersion": "2.10"
+    }
+  ],
+  "generated": "2022-01-01T00:00:00.000000Z",
+  "generator": "pub",
+  "generatorVersion": "2.17.3"
+}
+''';
```

Comment:
Consider:
```suggestion
  final configTemplate = json.encode({
  "configVersion": 2,
  "packages": [
    {
      "name": "test_package",
      "rootUri": "../",
      "packageUri": "lib/",
      "languageVersion": "2.10"
    }
  ],
  "generated": "2022-01-01T00:00:00.000000Z",
  "generator": "pub",
  "generatorVersion": "2.17.3"
});
```

Reply:
Also consider not using language version 2.10 anywhere... ensure that lower-bound SDK constraint is always `>= 2.12`, so that you've opted into null-safety.

Reply:
alright! not sure why languageVersion isn't the same as generatorVersion here, though - I've had Dart 2.17.3 installed on my machine throughout

Reply:
See: https://dart.dev/guides/language/evolution#language-versioning